core/arch/arm/kernel/generic_boot.c

```diff
@@ -484,6 +484,56 @@ static void reset_dt_references(void)
 	dt_blob_addr = NULL;
 }
 
+#ifdef CFG_DT_OVERLAY
+static int __dt_frag_id;
```

Comment:
The value of this variable is only valid together with `dt_blob_addr`
To make the connection more clear I think they should be declared together.

Ideally, if it works I'd like to keep them in a struct together. Instead of passing the fdt pointer to `add_optee_dt_node()` and `config_psci()` a pointer to this new struct could be passed.

Reply:
OK.
I added this change on as a separate patch at the end.